### ISSUE TITLE
can I use .guff

### ISSUE THREAD

**Issue by krecicki** [ronith256/LocalGPT-Android#3]
I went through the code and it downloads a model from GPT4ALL.

How can I add my .guff file to the android project and use it instead? I won't be able to share it on the App Store but thats OK.

How can I update this?

**Comment by ronith256** [comment]
This is using the gptj library to execute the model. To use gguf models, you'll have to use the llamacpp inference library compiled for Android. I'm working on a seperate app which will let you inference any gguf model using llamacpp for android. But that will take sometime as I'm a college student :)

**Comment by ayush20501** [comment]
> This is using the gptj library to execute the model. To use gguf models, you'll have to use the llamacpp inference library compiled for Android. I'm working on a seperate app which will let you inference any gguf model using llamacpp for android. But that will take sometime as I'm a college student :)

Are you able to built the app?

**Comment by ronith256** [comment]
> > This is using the gptj library to execute the model. To use gguf models, you'll have to use the llamacpp inference library compiled for Android. I'm working on a seperate app which will let you inference any gguf model using llamacpp for android. But that will take sometime as I'm a college student :)
> 
> Are you able to built the app?

Yes, I have a working model but it's still WIP. I am busy with college and actual work now. I don't think I'll be working on this project for a few months.

**Comment by altkriz** [comment]
> I went through the code and it downloads a model from GPT4ALL.
> 
> How can I add my .guff file to the android project and use it instead? I won't be able to share it on the App Store but thats OK.
> 
> How can I update this?

i think there is another app named chatterui if you want to use gguf models it works perfectly for gguf i tried it myself on my phone , phone should have a powerful processor like snapdragon  or exynos 2200 or above and empty ram 4GB
